src/app/routes/ssoRoutes.js

```diff
@@ -423,8 +423,8 @@ module.exports = (app) => {
     }
   });
 
-  app.all('/get/user/sessionId', (req, res) => {
-      res.send({id: getEncyptedQueryParams({userName: req.query.userName})})
+  app.all('/learner/get/user/sessionId', (req, res) => {
+      res.send({id: getEncyptedQueryParams({userName: req.session.userName})})
```

Comment:
This is not going to solve the problem, You have to take userId as input & compare with session userId. If it matches then send the userName else through error

Reply:
this way we will secure this endpoint. Other users can;t take someother users token.